presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestingPrometheusServer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.spi.type.TypeManager;
+import io.prestosql.type.InternalTypeManager;
+import org.testcontainers.containers.GenericContainer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.plugin.prometheus.MetadataUtil.METRIC_CODEC;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Integration test against Prometheus container
+ */
+public class TestingPrometheusServer
+{
+    private static GenericContainer prometheus = new GenericContainer("prom/prometheus:v2.15.1").withExposedPorts(9090);
```

Comment:
We would like to use a docker container (testcontainers) for end-to-end test. You can see `MongoServer` and `MongoQueryRunner` as a reference.

Reply:
Many tests are now using testcontainer docker container. I can split the TestingPrometheusServer class into container starter and query running test classes if desired like Mongo, but it seems okay as-is too.